course3/UniquePtr/ptr_test.cc

```diff
@@ -0,0 +1,73 @@
+#include "my_unique_ptr.h"
+#include <array>
+#include <string>
+#include <cassert>
+#include <iostream>
+#include <cstring>
+#include <memory>
+
+
+struct Foo
+{
+    Foo()      { std::cout << "Foo::Foo\n";  }
+    ~Foo()     { std::cout << "Foo::~Foo\n"; }
+    void bar() { std::cout << "Foo::bar\n";  }
+    int ret6() { std::cout << "Foo::ret6\n"; return 6; }
+    std::array<int, 5> retArray(){
+        std::array<int, 5> arr;
+        for(int i=0; i<5; ++i) {
+            arr[i]=i*2;
+        }
+    return arr;
+    }
+};
+
+
+std::string s(const Foo &){
+    std::cout << "s(const Foo&)\n";
+    return "aString";
+}
+
+
+int main()
+{
+    
+    std::cout << "Testing std::unique_ptr..." << std::endl;
+    std::unique_ptr<Foo> std_p1(new Foo);  // p1 owns Foo
+    std::unique_ptr<Foo> std_p2(std::move(std_p1));  // now p2 owns Foo
+    assert(std_p1->ret6() == std_p2->ret6());
```

Comment:
Here you are calling a method of a deleted object. This doesn't generate any error, because the method doesn't actually accesses any of the object's data.
Why do you expect this test to be true? std_p1 should be in a "zombie" state after being "moved from"